src/test/scala/io/iohk/ethereum/faucet/jsonrpc/WalletServiceSpec.scala

```diff
@@ -51,15 +37,56 @@ class FaucetRpcServiceSpec
 
     val retTxId = ByteString(Hex.decode("112233"))
 
-    (mockKeyStore.unlockAccount _).expects(config.walletAddress, config.walletPassword).returning(Right(wallet))
     (mockRpcClient.getNonce _).expects(config.walletAddress).returning(Right(currentNonce))
     (mockRpcClient.sendTransaction _).expects(ByteString(expectedTx)).returning(Right(retTxId))
 
-    val faucetRpcService = new FaucetRpcService(mockRpcClient, mockKeyStore, config)
+    val res = walletService.sendFunds(wallet, Address("0x99")).runSyncUnsafe()
+
+    res shouldEqual Right(retTxId)
+
+  }
+
+  it should "get wallet successful" in new TestSetup {
+    (mockKeyStore.unlockAccount _).expects(config.walletAddress, config.walletPassword).returning(Right(wallet))
 
-    val res = faucetRpcService.sendFunds(SendFundsRequest(Address("0x99"))).runSyncUnsafe()
+    val res = walletService.getWallet.runSyncUnsafe()
 
-    res shouldEqual Right(SendFundsResponse(retTxId))
+    res shouldEqual Right(wallet)
+  }
+
+  it should "wallet decryption failed" in new TestSetup {
+    (mockKeyStore.unlockAccount _)
+      .expects(config.walletAddress, config.walletPassword)
+      .returning(Left(DecryptionFailed))
+
+    val res = walletService.getWallet.runSyncUnsafe()
+
+    res shouldEqual Left(DecryptionFailed)
+  }
+
+  trait TestSetup {
+    val walletKeyPair = generateKeyPair(new SecureRandom)
+    val (prvKey, pubKey) = keyPairToByteStrings(walletKeyPair)
+    val wallet = Wallet(Address(crypto.kec256(pubKey)), prvKey)
+
+    val mockRpcClient = mock[RpcClient]
+    val mockKeyStore = mock[KeyStore]
+    val config: FaucetConfig =
+      FaucetConfig(
```

Comment:
^ far more readable if someone is using just github to check the code, or is in a server environment.

Reply:
Done